packages/slate-react/src/plugin/react-editor.ts

```diff
@@ -111,9 +113,7 @@ export const ReactEditor = {
     if (el.ownerDocument !== document) return el.ownerDocument
 
     if (!(root instanceof Document || root instanceof ShadowRoot))
-      throw new Error(
-        `Unable to find DocumentOrShadowRoot for editor element: ${el}`
-      )
+      return undefined
```

Comment:
Looking back at https://github.com/ianstormtaylor/slate/pull/3749/files , I think this may just suppress things, rather than restoring support for window/window.document. Perhaps better would be:

```suggestion
      return ReactEditor.getWindow(editor).document
```

And then we wouldn't need the change to support `| undefined` throughout? Unless I'm missing something (e.g. if window wasn't actually working previously).

Reply:
This is the correct fix given what used to happen before #3749, and after it's merged every other change (to deal with potentially undefined return value) can be reverted.

I also suggest mentioning in the changeset that this fixes a regression caused by #3749.

Reply:
Good to go now!